Tests/GitHubCodespaces.tests.ps1

```diff
@@ -0,0 +1,459 @@
+param()
+
+# This is common test code setup logic for all Pester test files
+BeforeAll {
+    $moduleRootPath = Split-Path -Path $PSScriptRoot -Parent
+    . (Join-Path -Path $moduleRootPath -ChildPath 'Tests\Common.ps1')
+
+    # Define Script-scoped, readonly, hidden variables.
+    @{
+        defaultRepositoryName = ([Guid]::NewGuid().Guid)
+    }.GetEnumerator() | ForEach-Object {
+        Set-Variable -Force -Scope Script -Option ReadOnly -Visibility Private -Name $_.Key -Value $_.Value
+    }
+
+    $newGitHubRepositoryParms = @{
+        AutoInit = $true
+        Private = $true
+        RepositoryName = $defaultRepositoryName
+        OrganizationName = $script:organizationName
+    }
+    $repo = New-GitHubRepository @newGitHubRepositoryParms
+    # Get the main/master branch name
+    $mainBranchName = $repo | Get-GitHubRepositoryBranch | Select-Object -ExpandProperty name -First 1
+}
+
+Describe 'GitHubCodespaces\Delete-GitHubCodespace' {
+    Context 'When deleting a codespace for the authenticated user' {
+        BeforeEach {
+            # Suppress HTTP 202 warning for codespace creation
+            # TODO: Suppression is not working as intended here.
+            $WarningPreference = 'SilentlyContinue'
+
+            $newGitHubCodespaceParms = @{
+                OwnerName = $script:organizationName
+                RepositoryName = $defaultRepositoryName
+                Wait = $true
+            }
+            $codespace = New-GitHubCodespace @newGitHubCodespaceParms
+        }
+
+        It 'Should get no content using -Confirm:$false' {
+            # Also asserts pipeline input
+            $codespace | Remove-GitHubCodespace -Confirm:$false
+            { Get-GitHubCodespace -CodespaceName $codespace.name } | Should -Throw
+        }
+
+        It 'Should get no content using -Force' {
+            # Also assert CodespaceName input
+            Remove-GitHubCodespace -CodespaceName $codespace.name -Force
+            { Get-GitHubCodespace -CodespaceName $codespace.name } | Should -Throw
+        }
+    }
+}
+
+Describe 'GitHubCodespaces\Get-GitHubCodespace' {
+    BeforeAll {
+        # Suppress HTTP 202 warning for codespace creation
+        $WarningPreference = 'SilentlyContinue'
+
+        $newGitHubCodespaceParms = @{
+            OwnerName = $script:organizationName
+            RepositoryName = $defaultRepositoryName
+            Wait = $true
+        }
+        $null = New-GitHubCodespace @newGitHubCodespaceParms
+    }
+
+    Context 'When getting codespaces for the authenticated user' {
+        BeforeAll {
+            $codespaces = Get-GitHubCodespace |
+                Where-Object { $_.repository.name -eq $defaultRepositoryName }
+        }
+
+        It 'Should return objects of the correct type' {
+            $codespaces[0].PSObject.TypeNames[0] | Should -Be 'GitHub.Codespace'
+        }
+
+        It 'Should return one or more results' {
+            $codespaces.Count | Should -BeGreaterOrEqual 1
```

Comment:
This is failing.  `$codespaces` is coming back as a `PSCustomObject`, not a `PSCustomObject[]`, so there is no `Count` on it.

``
 Expected the actual value to be greater than or equal to 1, but got $null.
``